include/stl2/view/const.hpp

```diff
@@ -0,0 +1,276 @@
+		{
+			current_ += n;
+			return *this;
+		}
+
+		constexpr __iterator &operator-=(difference_type const n) requires random_access_range<base_t>
+		{
+			current_ -= n;
+			return *this;
+		}
+
+		constexpr decltype(auto) operator[](difference_type const n) const
+				requires random_access_range<base_t>
+		{
+			return *(*this + n);
+		}
+
+		friend constexpr bool operator==(const __iterator &x, const __iterator &y) requires equality_comparable<iterator_t<base_t>>
+		{
+			return x.current_ == y.current_;
+		}
+
+		friend constexpr bool operator!=(const __iterator &x, const __iterator &y) requires equality_comparable<iterator_t<base_t>>
+		{
+			return !(x == y);
+		}
+
+		friend constexpr bool operator==(const __iterator &x, const sentinel_t<base_t> &y)
+		{
+			return x.current_ == y;
+		}
+
+		friend constexpr bool operator==(const sentinel_t<base_t> &y, const __iterator &x)
+		{
+			return x == y;
+		}
+
+		friend constexpr bool operator!=(const __iterator &x, const sentinel_t<base_t> &y)
+		{
+			return !(x == y);
+		}
+
+		friend constexpr bool operator!=(const sentinel_t<base_t> &y, const __iterator &x)
+		{
+			return !(x == y);
+		}
+
+		friend constexpr bool operator<(const __iterator &x, const __iterator &y) requires random_access_range<base_t>
+		{
+			return x.current_ < y.current_;
+		}
+
+		friend constexpr bool operator>(const __iterator &x, const __iterator &y) requires random_access_range<base_t>
+		{
+			return y < x;
+		}
+
+		friend constexpr bool operator<=(const __iterator &x, const __iterator &y) requires random_access_range<base_t>
+		{
+			return !(y < x);
+		}
+
+		friend constexpr bool operator>=(const __iterator &x, const __iterator &y) requires random_access_range<base_t>
+		{
+			return !(x < y);
+		}
+
+		friend constexpr __iterator operator+(__iterator i, difference_type n) requires random_access_range<base_t>
+		{
+			return i += n;
+		}
+
+		friend constexpr __iterator operator+(difference_type n, __iterator i) requires random_access_range<base_t>
+		{
+			return i += n;
+		}
+
+		friend constexpr __iterator operator-(__iterator i, difference_type n) requires random_access_range<base_t>
+		{
+			return i -= n;
+		}
+
+		friend constexpr difference_type operator-(const __iterator &x, const __iterator &y) requires random_access_range<base_t>
+		{
+			return x.current_ - y.current_;
+		}
+
+		friend constexpr difference_type operator-(const __iterator &x, const sentinel_t<base_t> &y) requires sized_sentinel_for<iterator_t<base_t>, sentinel_t<base_t>>
+		{
+			return x.current_ - y;
+		}
+
+		friend constexpr difference_type operator-(const sentinel_t<base_t> &x, const __iterator &y) requires sized_sentinel_for<iterator_t<base_t>, sentinel_t<base_t>>
+		{
+			return x - y.current_;
+		}
+	};
+
+	template <class R>
+	const_view(R &&r)->const_view<all_view<R>>;
+
+	} // namespace ext
+
+	namespace views::ext
+	{
+	struct __as_const_fn : detail::__pipeable<__as_const_fn>
```

Comment:
Inheritance not needed for this one. Also, please look at some of the other adaptors to see what's usually done. I recommend `take` or `drop`.

Reply:
I think it's necessary for adaptors with no arguments. same as `views::move`.